gunicorn/reloader.py

```diff
@@ -105,7 +105,7 @@ def run(self):
             self._dirs = self.get_dirs()
 
             for dirname in self._dirs:
-                self._watcher.add_watch(dirname, mask=self.event_mask)
```

Comment:
It doesn't feel great calling a method that looks private. I noticed that `InotifyTrees` takes a list of paths in its constructor. Can we defer instantiation until this `run` method and pass `self._dirs` to the constructor, or do we need `_watcher` to exist earlier for some reason?

Reply:
If we want to support `add_extra_file` then we will have to call  `__load_trees` after instantiation. So basically even if I defer to the constructor in the `run` method at some point we will still need to call `__load_trees`.

We can avoid calling this method if we copy paste its logic but I think in this case it should be fine.

Reply:
I'm not sure there's any reason we don't defer calling `self.reloader.start()` in `gunicorn.workers.base` until after `self.load_wsgi()`. If we defer starting the reloader we can maybe assume that all files are added by the time we call `self.reloader.start()`.

I wonder whether the use case of adding extra files after starting the application is needed.

Reply:
We don't need to reload extra files after startup and your proposed solution would work for us. So I think that is the better way to go.